engine/src/retrier.rs

```diff
@@ -297,6 +308,20 @@ impl<Client: Send + Sync + Clone + 'static> ClientSelector<Client> {
 		.await
 		.0
 	}
+
+	pub fn request_failed(&mut self, failed_client: PrimaryOrSecondary) {
+		// If we don't have a second endpoint, then we can only try the primary.
+		if self.secondary_signal.is_none() {
+			return;
+		}
+
+		self.prefer = if failed_client == PrimaryOrSecondary::Primary {
+			self.last_tried_primary = Some(tokio::time::Instant::now());
```

Comment:
tried is wrong, the time is when the primary last "failed".

Reply:
guess it's last "tried" assuming we're currently using the backup - but yeah failed is more explicit, will change